cmd/celestia/share.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	"github.com/celestiaorg/celestia-app/pkg/da"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+func init() {
+	shareCmd.AddCommand(
+		sharesAvailableCmd,
+		probabilityOfAvailabilityCmd,
+		getSharesByNamespaceCmd,
+		getShare,
+		getEDS,
+	)
+}
+
+var shareCmd = &cobra.Command{
+	Use:   "share [command]",
+	Short: "Allows interaction with the Share Module via JSON-RPC",
+	Args:  cobra.NoArgs,
+}
+
+var sharesAvailableCmd = &cobra.Command{
+	Use:   "shares-available",
+	Short: "Subjectively validates if Shares committed to the given Root are available on the Network.",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		raw, err := parseJSON(args[0])
+		if err != nil {
+			return err
+		}
+
+		root := da.MinDataAvailabilityHeader()
+		err = json.Unmarshal(raw, &root)
+		if err != nil {
+			return err
+		}
+
+		err = client.Share.SharesAvailable(cmd.Context(), &root)
+		formatter := func(data interface{}) interface{} {
+			err, ok := data.(error)
+			available := false
+			if !ok {
+				available = true
+			}
+			return struct {
+				Available bool   `json:"available"`
+				Hash      []byte `json:"dah_hash"`
+				Reason    error  `json:"reason,omitempty"`
+			}{
+				Available: available,
+				Hash:      []byte(args[0]),
+				Reason:    err,
+			}
+		}
+		return printOutput(err, nil, formatter)
+	},
+}
+
+var probabilityOfAvailabilityCmd = &cobra.Command{
```

Comment:
It's a feature request we had a while ago that we implemented quickly in the API. We still need it in some form or another. It should be merged with `ShareAvailable` call so that we don't just return binary answers, but a percentage of guaranteed local availability.

I suggest keeping the current cmd and merging it once we merge SharesAvailable API.

Reply:
I agree, we can merge it as it is and then remove it when it becomes redundant.

Reply:
Or I can return the percentage w SharesAvaliable(so `available` cmd will make 2 rpc call now)

Reply:
> Wish there was an easy way to write unit tests for these

Tested all cmds on arabica10